### PR TITLE
Website: Product Configurator and Bootstrap 4

### PR DESCRIPTION
**Pull request by netchampfaris**
- Banner Image
- Add item meta tags
- Product page redesign

depends on https://github.com/frappe/frappe/pull/6872

image

### CODE REVIEW

File: erpnext/templates/generators/item.html [hunk 1]
```diff
@@ -11,133 +14,212 @@
...
+								</div>
+							</td>
+							<td><button data-item-code="{{item_variant.name}}" class="btn btn-outline-primary item-variant-add-to-cart">Add to Cart</button></td>
+						</tr>
+						{% endfor %}
+					</tbody>
+				</table>
+			{% endif %}
+		</div>
+		<!-- <div class="item-attribute-selectors">
```
Comment (rmehta) [inline erpnext/templates/generators/item.html]: remove commented code

**Review by rmehta — DISMISSED** [review]
add a few basic tests

**Review by scmmishra — DISMISSED** [review]
- [x] Only show published items on /products
- [x] Empty state handling for homepage
- [x] Homepage Section is not discoverable
- [x] Handle image sizes better on /products

**Comment by netchampfaris** [comment]
Codacy is picking up someone else's url, it passed in frappe's [codacy](https://app.codacy.com/app/frappe/erpnext/pullRequest?prid=2540357).

**Comment by jigneshpshah** [comment]
Error on website for new Item not having image and web_long_description .

#### Environment :
ERPNext: v12.0.4 (version-12)
Frappe Framework: v12.0.4 (version-12) 

#### Step To Reproduce : 

1. Create an Item 
2. Check -> Show in Website
3. Click on "See on Website" link

### Actual Behavior 

System generates below error

Uncaught Server Exception
```
Traceback (most recent call last):
  File "/home/frappe/frappe-bench/apps/frappe/frappe/website/render.py", line 48, in render
    data = render_page_by_language(path)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/website/render.py", line 152, in render_page_by_language
    return render_page(path)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/website/render.py", line 168, in render_page
    return build(path)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/website/render.py", line 175, in build
    return build_page(path)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/website/render.py", line 197, in build_page
    html = frappe.get_template(context.template).render(context)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/./templates/generators/item/item.html", line 1, in top-level template code
    {% extends "templates/web.html" %}
  File "/home/frappe/frappe-bench/apps/frappe/frappe/./templates/web.html", line 1, in top-level template code
    {% extends base_template_path %}
  File "/home/frappe/frappe-bench/apps/frappe/frappe/./templates/base.html", line 68, in top-level template code
    {% block content %}
  File "/home/frappe/frappe-bench/apps/frappe/frappe/./templates/web.html", line 60, in block "content"
    {{ main_content() }}
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/sandbox.py", line 438, in call
    return __context.call(__obj, *args, **kwargs)
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/runtime.py", line 579, in _invoke
    rv = self._func(*arguments)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/./templates/web.html", line 15, in template
    {% block page_container %}
  File "/home/frappe/frappe-bench/apps/frappe/frappe/./templates/web.html", line 30, in block "page_container"
    {%- block page_content -%}{%- endblock -%}
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/./templates/generators/item/item.html", line 15, in block "page_content"
    {% include "templates/generators/item/item_details.html" %}
  File "/home/frappe/frappe-bench/apps/erpnext/erpnext/./templates/generators/item/item_details.html", line 12, in top-level template code
    {% if frappe.utils.strip_html(doc.web_long_description) %}
  File "/home/frappe/frappe-bench/env/lib/python3.6/site-packages/jinja2/sandbox.py", line 438, in call
    return __context.call(__obj, *args, **kwargs)
  File "/home/frappe/frappe-bench/apps/frappe/frappe/utils/data.py", line 607, in strip_html
    return _striptags_re.sub("", text)
TypeError: expected string or bytes-like object

```

4. Put some text in web_long_description , while keeping image blank,  it will show below error

image

### Expected Behavior 
For item not having image and web_long_description , system should not generate error and should display the item page normally

Note : For the above scenario, it worked fine in version 11 , without any issue.
ERPNext: v11.1.59 (version-11)
Frappe Framework: v11.1.54 (version-11)

**Pull request merged**